### PR TITLE
feat: upgraded to node v18, added .nvmrc and updated workflows

### PR DESCRIPTION
**Pull request by Mashal-m**
- Added support for node v18, updated .nvmrc and workflows.
- Updated frontend-platform, frontend-build, frontend-component-header, paragon, and frontend-component-footer versions.

**Ticket**
[Upgrade Node JS from 16 to 18](https://github.com/openedx/frontend-app-ora-grading/issues/173)

**edx-internal PR**

https://github.com/edx/edx-internal/pull/8329

This needs to be merged together.

### CODE REVIEW

File: src/data/constants/app.test.js [hunk 1]
```diff
@@ -17,8 +17,8 @@ describe('app constants', () => {
   });
   test('locationId returns trimmed pathname', () => {
     const old = window.location;
-    window.location = { pathName: '/somePath.jpg' };
+    window.location ??= { pathName: '/somePath.jpg' };
```
Comment (leangseu-edx) [inline src/data/constants/app.test.js]: This changed the intention of the previous code.

Comment (Mashal-m) [reply src/data/constants/app.test.js]: Had to add this to fix failing tests.

Comment (leangseu-edx) [reply src/data/constants/app.test.js]: It seems that this is a bug on node 18. https://github.com/nodejs/node/issues/47563 . I don't like the solution to force this code in right now then remember to remove it later. I would rather a wait until the next release of node `18.x` and not change this test.

Comment (muselesscreator) [reply src/data/constants/app.test.js]: ^

Comment (Mashal-m) [reply src/data/constants/app.test.js]: I have pinned the node version to **18.15** until the next release.
@leangseu-edx

File: .nvmrc [hunk 2]
```diff
@@ -0,0 +1 @@
+18
```
Comment (leangseu-edx) [inline .nvmrc]:
```suggestion
18.15
```

**Comment by leangseu-edx** [comment]
One requested change. I will approve once this get update.

**Comment by abdullahwaheed** [comment]
> One requested change. I will approve once this get update.

@leangseu-edx we have pinned down node version and updated code according to your suggestion. could you please review and merge this

**Comment by leangseu-edx** [comment]
> > One requested change. I will approve once this get update.
> 
> @leangseu-edx we have pinned down node version and updated code according to your suggestion. could you please review and merge this

Should we include that version `.nvmrc` as well?

**Review by leangseu-edx — APPROVED** [review]

**Pull request merged**